### PR TITLE
Add skip error option

### PR DESCRIPTION
**Pull request by uesyn**
I created a patch which adds a option to skip error.

### CODE REVIEW

File: patches/add-skip-error-opition.patch [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
++partially_failed=0
+ 
+ for resource in "${resources[@]}"
+ do
+@@ -118,7 +121,8 @@ do
+       if [[ "${success}" -eq 0 ]]
+       then
+         echo "Error: failed to update ${resource}/${instance} in ${namespace} namespace after 5 tries"
+-        exit 1
++        [[ ${SKIP_UPDATE_OBJECT_ERROR} =~ ^(t|true|TRUE|y|yes|YES)$ ]] || exit 1
```
Comment (yuanying) [inline patches/add-skip-error-opition.patch]: IHMO, using regexp can be avoided. How about checking simply empty string or not.

Comment (superbrothers) [reply patches/add-skip-error-opition.patch]: I have the same opinion.

Comment (uesyn) [reply patches/add-skip-error-opition.patch]: I corrected the points that were pointed out.

**Review by yuanying — APPROVED** [review]

File: patches/add-skip-error-opition.patch [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
++partially_failed=0
+ 
+ for resource in "${resources[@]}"
+ do
+@@ -118,7 +121,8 @@ do
+       if [[ "${success}" -eq 0 ]]
+       then
+         echo "Error: failed to update ${resource}/${instance} in ${namespace} namespace after 5 tries"
+-        exit 1
++        [[ -n ${SKIP_UPDATE_OBJECT_ERROR} ]] || exit 1
```
Comment (superbrothers) [inline patches/add-skip-error-opition.patch]:
```suggestion
+        [[ -n "${SKIP_UPDATE_OBJECT_ERROR}" ]] || exit 1
```

File: patches/add-skip-error-opition.patch [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
++        partially_failed=1
+       fi
+     done
+     if [[ "${resource}" == "namespaces" ]] || [[ "${resource}" == "nodes" ]]
+@@ -130,6 +134,11 @@ do
+   done
+ done
+ 
++if  [[ $partially_failed == 1 ]]; then
```
Comment (superbrothers) [inline patches/add-skip-error-opition.patch]:
```suggestion
+if  [[ "${partially_failed}" -eq 1 ]]; then
```

File: patches/add-skip-error-opition.patch [hunk 4]
```diff
@@ -0,0 +1,43 @@
+diff --git a/cluster/update-storage-objects.sh b/cluster/update-storage-objects.sh
+index d1f8affa13..5646abd776 100755
+--- a/cluster/update-storage-objects.sh
++++ b/cluster/update-storage-objects.sh
+@@ -25,6 +25,8 @@ set -o errexit
+ set -o nounset
+ set -o pipefail
+ 
++SKIP_UPDATE_OBJECT_ERROR=${SKIP_UPDATE_OBJECT_ERROR:-}
```
Comment (superbrothers) [inline patches/add-skip-error-opition.patch]:
```suggestion
+SKIP_UPDATE_OBJECT_ERROR="${SKIP_UPDATE_OBJECT_ERROR:-}"
```

**Pull request merged**